Shared/Extensions/UI Extensions/UIViewController+Extensions.swift

```diff
@@ -19,7 +19,12 @@ extension UIViewController {
         viewController.didMove(toParent: self)
     }
 
-    func removeFromParentSuperview() {
+    func removeFromParentAndSuperviewIfNeeded() {
+        guard self.parent.exists else {
+            self.view.removeFromSuperview()
+            return
+        }
```

Comment:
Adding "if needed" so we don't need to check if the view controller actually has a parent.